From d399f164beb6569ce627693b2527015f4e5c365a Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 2 Jan 2025 18:13:47 +0100
Subject: [PATCH 0138/1013] dovetail: net: rename oob_data to sk_oob_ctx in
 struct sock

The notion of "oob data" is used across several kernel subsystems,
particularly in the networking area. Dovetail's notion of out-of-band
relates to the execution context of a CPU on the high priority stage
of the interrupt pipeline instead (i.e. the one which runs
asynchronously to the regular scheduling of operations).

Mitigate the confusion in struct file by moving from the way too
generic "data" to "context", abiding by the common naming scheme there
in the same move, i.e. s/oob_data/sk_oob_ctx/.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/net/sock.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/include/net/sock.h b/include/net/sock.h
index 2cf4de1977f5..bea41e33375d 100644
--- a/include/net/sock.h
+++ b/include/net/sock.h
@@ -541,7 +541,7 @@ struct sock {
 	void                    (*sk_destruct)(struct sock *sk);
 	struct sock_reuseport __rcu	*sk_reuseport_cb;
 #ifdef CONFIG_NET_OOB
-	void			*oob_data;
+	void			*sk_oob_ctx;
 #endif
 #ifdef CONFIG_BPF_SYSCALL
 	struct bpf_local_storage __rcu	*sk_bpf_storage;
@@ -1808,7 +1808,7 @@ void sock_pfree(struct sk_buff *skb);
 
 static inline bool sock_oob_capable(struct socket *sock)
 {
-	return sock && sock->sk && sock->sk->oob_data;
+	return sock && sock->sk && sock->sk->sk_oob_ctx;
 }
 
 int sock_oob_attach(struct socket *sock);
@@ -1844,7 +1844,8 @@ __poll_t sock_oob_poll(struct file *filp,
 
 static inline void sock_oob_destruct(struct sock *sk)
 {
-	if (sk->oob_data)
+	void sock_oob_destroy(struct sock *sk);
+	if (sk->sk_oob_ctx)
 		sock_oob_destroy(sk);
 }
 
-- 
2.30.2

